src/durableorchestrationcontext.ts

```diff
@@ -571,46 +253,22 @@ export class DurableOrchestrationContext {
 
         const req = new DurableHttpRequest(method, uri, content as string, headers, tokenSource);
         const newAction = new CallHttpAction(req);
-
-        // callHttp is internally implemented as a well-known activity function
-        const httpScheduled = this.findTaskScheduled(this.state, "BuiltIn::HttpActivity");
-        const httpCompleted = this.findTaskCompleted(this.state, httpScheduled);
-        const httpFailed = this.findTaskFailed(this.state, httpScheduled);
-        this.setProcessed([httpScheduled, httpCompleted, httpFailed]);
-
-        if (httpCompleted) {
-            const result = this.parseHistoryEvent(httpCompleted);
-
-            return TaskFactory.SuccessfulTask(
-                newAction,
-                result,
-                httpCompleted.Timestamp,
-                httpCompleted.TaskScheduledId,
-                this.state.indexOf(httpCompleted)
-            );
-        } else if (httpFailed) {
-            return TaskFactory.FailedTask(
-                newAction,
-                httpFailed.Reason,
-                httpFailed.Timestamp,
-                httpFailed.TaskScheduledId,
-                this.state.indexOf(httpFailed),
-                new DurableError(httpFailed.Reason)
-            );
-        } else {
-            return TaskFactory.UncompletedTask(newAction);
-        }
+        const innerTask = new AtomicTask(false, newAction);
+        const task = new Task(innerTask);
+        return task;
     }
 
     /**
      * Restarts the orchestration by clearing its history.
      *
      * @param The JSON-serializable data to re-initialize the instance with.
      */
-    public continueAsNew(input: unknown): Task {
+    public continueAsNew(input: unknown): void {
         const newAction = new ContinueAsNewAction(input);
-
-        return TaskFactory.UncompletedTask(newAction);
+        if (this.taskOrchestratorExecutor !== undefined) {

```

Comment:
I believe it's undefined during testing, mostly due to how old tests were implemented. I could try to refactor those old tests, but when first implementing this PR I thought it would be too much work. I can reconsider this decision if we think it is a big deal though :)

Reply:
Definitely not a big deal. A comment might be good, though, to help maintainers understand that this is not a normal scenario that needs special handling.

Reply:
I ended up removing this check, as apparently no tests broke without it